fusilli-plugin/build_tools/cmake/FusilliPluginDependencyUtils.cmake

```diff
@@ -0,0 +1,204 @@
+# Copyright 2025 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#===----------------------------------------------------------------------===#
+#
+# Provides correctly configured dependencies for fusilli-plugin build.
+#
+# Main entry point:
+#   fusilli_plugin_dependency(DEP_NAME [args...])
+#
+# `fusilli_plugin_dependency` routes to lower level `_fetch_X` macros to
+# actually fetch dependency `X`. Each `_fetch_X` macro preferentially
+# `find_package`s installed/system versions of packages and falls back to
+# vendoring dependencies in the build tree with `FetchContent`.
+#
+# Supported dependencies: GTest, hipdnn_frontend, Fusilli, IREERuntime
+#
+#===----------------------------------------------------------------------===#
+
+cmake_minimum_required(VERSION 3.25.2)
+
+include(FetchContent)
+
+# Provide a fusilli plugin dependency. `fusilli_plugin_dependency` will
+# preferentially use system version (available through `find_package`) of a
+# dependency, and fall back to building local copy with `FetchContent` +
+# configuration.
+#
+#  fusilli_plugin_dependency(
+#    DEP_NAME
+#    [<dependency-specific args>...]
+#  )
+#
+# DEP_NAME
+#   Supported dependencies:
+#     GTest
+#     hipdnn_frontend
+#     Fusilli
+#     IREERuntime
+#
+# <dependency-specific args>
+#   The `_fetch_X` macro for dependency X defines the available options.
+#   Examples: GTEST_VERSION for GTest, HIP_DNN_HASH for hipdnn_frontend
+function(fusilli_plugin_dependency dep_name)
+    # Route to appropriate `_fetch_X` macro.
+    if(COMMAND _fetch_${dep_name})
+        _with_logging(_fetch_${dep_name} ${dep_name} ${ARGN})
+    else()
+        message(FATAL_ERROR "Unknown dependency: ${dep_name}")
+    endif()
+endfunction()
+
+# Think of this as a python decorator, wraps a `_fetch_X` macro with some
+# logging conveniences.
+#
+# MACRO_NAME
+#   Name of the `_fetch_X` macro.
+#   NOTE: CMake macros execute in current scope, CMake functions create a new
+#         scope. `_with_logging` checks variables it expects a `_fetch_X` macro
+#         to set _in_ `_with_logging`s scope, requiring that `_fetch_X` is a
+#         macro and not a function.
+#
+# DEP_NAME
+#   Dependency name for logging
+function(_with_logging MACRO_NAME DEP_NAME)
```

Comment:
I think I can collapse what this function is doing into `fusilli_plugin_dependency` actually. I don't think extra "hop" is necessary.